### DISCUSSION TITLE
WinUI3 unpackaged self contained application publishing is failing

### DISCUSSION THREAD

**Discussion by pbasithmpy** [microsoft/WindowsAppSDK#3026] [Q&A]
I was trying to make a winui3 unpackaged self contained application. But when I try to publish application with "dotnet publish" it is getting an error saying like this,

.nuget\packages\microsoft.windowsappsdk\1.1.5\buildTransitive\Microsoft.WindowsAppSDK.SelfContained.targ
ets(42,5): error : The platform 'AnyCPU' is not supported for Self Contained mode. [C:\Users\bsthp\source\repos\XSight_
windows\XSightUI\XSightUI.csproj]

command that I tried :  dotnet publish ..\UI\UI.csproj -c $(Configuration) -r win10-x64 -p:PublishSingleFile=true  -o ..\Executables\UI

**Comment by beeradmoore** [comment]
For some reason platform needs to be defined as well. With this included on the command you are running it would now look like

```
dotnet publish ..\UI\UI.csproj -c $(Configuration) -r win10-x64 -p:PublishSingleFile=true -p:Platform=x64 -o ..\Executables\UI
```

**Reply by gary-89** [reply]
This helped me solving the issue I have with "The platform 'AnyCPU' is not supported for Self Contained mode.". 
Just add the platform parameter to your command with `-p:Platform=...`
Thanks.
Maybe this solution can helpful for: #2684 #3166
